service/app.py

```diff
@@ -0,0 +1,103 @@
+import os
+import subprocess
+import uuid
+
+from flask import Flask
+from flask_restful import Resource, Api, reqparse
+
+import werkzeug
+
+# Set up the flask app and API
+app = Flask(__name__)
+api = Api(app)
+
+# Declare two global variables that will hold state.
+WORKFLOW_TESTS = {}
+ENVIRONMENTS = {}
+
+# Functions for running the test
+def submit_test(environment_image, workflow_source, workflow_inputs, workflow_dependencies):
+
+    docker_cmd = ["docker", "run", "-v /io:/io", "-v /var/run/docker.sock:/var/run/docker.sock", environment_image]
+    docker_cmd.extend(["--workflowSource", workflow_source])
+    docker_cmd.extend(["--workflowInputs", workflow_inputs])
+    if workflow_dependencies:
+        docker_cmd.extend(["--workflowDependencies", workflow_dependencies])
+
+    proc = subprocess.Popen(docker_cmd)
+    proc.communicate()
+
+
+    if proc.returncode == 0:
+        workflow_test = {"status": "Succeeded"}
+    else:
+        workflow_test = {"status": "Failed"}
+    workflow_test["command"] = ' '.join(docker_cmd)
+
+    return workflow_test
+
+class WorkflowTest(Resource):
+    def get(self):
+        parser = reqparse.RequestParser()
+        parser.add_argument("workflowTestId")
+        args = parser.parse_args()
+        return WORKFLOW_TESTS[args.WorkflowTestId]
```

Comment:
might be good to try/except here on `KeyError` and explain that the workflow doesn't exist.

Reply:
It looks like you're supposed to use `flask_restful.abort(404, ...)` for something like this, so I'll switch to that.